src/Stream/src/Abstractions/Binder/IPartitionKeyExtractorStrategy.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Steeltoe.Messaging;
+
+namespace Steeltoe.Stream.Binder
+{
+    /// <summary>
+    /// Strategy for extracting a partition key from a Message.
+    /// </summary>
+    public interface IPartitionKeyExtractorStrategy
+    {
+        /// <summary>
+        /// Gets the name of the strategy
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// Extract the partition key from the incoming message
+        /// </summary>
+        /// <param name="message">the message to process</param>
+        /// <returns>the key</returns>
+        object ExtractKey(IMessage message);
```

Comment:
should this just return `string` instead of `object`?

Reply:
Java code has it an object ..